src/amo/components/AutoSearchInput/index.js

```diff
@@ -154,30 +155,35 @@ export class AutoSearchInputBase extends React.Component<Props, State> {
   }
 
   handleSuggestionsFetchRequested = (
-    { value, reason }: OnSuggestionsFetchRequestedParams
+    { value }: OnSuggestionsFetchRequestedParams
   ) => {
     if (!value) {
       log.debug(oneLine`Ignoring suggestions fetch requested because
         value is not supplied: ${value}`);
       return;
     }
 
+    if (value.length < SEARCH_TERM_MIN_LENGTH) {
+      log.debug(oneLine`Ignoring suggestions fetch because query
+      does not meet the required length (${SEARCH_TERM_MIN_LENGTH})`);
+
+      this.props.dispatch(autocompleteCancel());
+      return;
+    }
+
     if (value.length > SEARCH_TERM_MAX_LENGTH) {
       log.debug(oneLine`Ignoring suggestions fetch because query
         exceeds max length (${SEARCH_TERM_MAX_LENGTH})`
       );
+
+      this.props.dispatch(autocompleteCancel());
       return;
     }
 
     const filters = this.createFiltersFromQuery(value);
 
     this.setState({ autocompleteIsOpen: true });
 
-    if (reason === 'input-focused') {
```

Comment:
Yep, I think this is the right thing to do given that we fixed the behavior of `handleSuggestionsClearRequested()`. @willdurand maybe you remember why we originally needed this? We can always put it back with a follow-up patch.